### PR TITLE
[Filebeat] Add support for Cloudtrail digest files

### LINKED ISSUE #20943: [Filebeat] AWS cloudtrail module should handle Digest files
**Describe the enhancement:**

AWS Cloudtrail can be configured to create a digest of each cloudtrail
log file when it writes it.  Filebeat should be able to ingest these
files.

https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-log-file-validation-intro.html

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

Adds support for Cloudtrail Digest Logs and Cloudtrail Insight logs.
Also adds options to ignore cloudtrail, cloudtrail-digest or
cloudtrail-insight logs if necessary.

## Why is it important?

- Cloudtrail digest files are used to determine integrity of Cloudtrail
logs.
- Users need the ability to ignore the digest or insight logs if necessary.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

```
TESTING_FILEBEAT_MODULES=aws TESTING_FILEBEAT_FILESETS=cloudtrail mage -v pythonIntegTest
```

## Related issues

- Closes #20943

### CODE REVIEW

File: x-pack/filebeat/input/s3/config.go [hunk 1]
```diff
@@ -19,6 +20,14 @@ type config struct {
...
 	ExpandEventListFromField  string              `config:"expand_event_list_from_field"`
 	APITimeout                time.Duration       `config:"api_timeout"`
+	FileSelectors             []FileSelectorCfg   `config:"file_selectors"`
+}
+
+// FileSelectorCfg defines type and configuration of FileSelectors
+type FileSelectorCfg struct {
+	RegexString              string         `config:"regex"`
+	Regex                    *regexp.Regexp `config:",ignore"`
+	ExpandEventListFromField string         `config:"expand_event_list_from_field"`
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/input/s3/config.go:30]: This looks good! One question: what happens if `file_selectors` are specified and also the global `expand_event_list_from_field` is given, which one takes priority? Maybe we should clarify this in the documentation 😄

File: filebeat/docs/fields.asciidoc [hunk 2]
```diff
@@ -1487,6 +1487,111 @@ type: flattened
...
+[float]
+=== digest
+
+Fields from Cloudtrail Digest Logs
+
+
+*`aws.cloudtrail.digest.log_files`*::
++
+--
+A list of Logfiles contained in the digest
```
Comment (kaiyan-sheng) [inline filebeat/docs/fields.asciidoc:1499]:
```suggestion
A list of Logfiles contained in the digest.
```

**Review by kaiyan-sheng — APPROVED** [review]
Thanks for adding this! It looks good overall besides we are missing a changelog entry here.

**Review by kaiyan-sheng — APPROVED** [review]

**Comment by leehinman** [comment]
jenkins run tests

**Pull request merged**